### DISCUSSION TITLE
How to add a custom PopupTitleTemplate for Add Edit on DataGrid

### DISCUSSION THREAD

**Discussion by CatalinCernea** [Megabit/Blazorise#2963] [Q&A]
I've been searching everywhere but found nothing.
How can I customize the title when the Add or Edit popup shows on the DataGrid?
Has anyone got an example?

**Comment by David-Moreira** [comment]
Hello take a look at `PopupTitleTemplate` it is a RenderFragment that provides you with the current popup context telling you the `DataGridEditState`, and you can do logic accordingly.

Here's an example, from the default implementation from our internal code:
![image](https://user-images.githubusercontent.com/22283161/134586263-717f6c05-5c58-4a89-b867-159b1f644b84.png)

PS: You don't need to do it like in the print, you can just use it like a regular RenderFragment in the Razor file Markup.

**Reply by sengkok** [reply]
For the beginner to easy understand how to add the edit model title , 
Add PopupTitleTemplate tag in the datagrid and the title name will do. 


```
<ButtonRowTemplate>
        <Button Color="Color.Success" Clicked="context.NewCommand.Clicked">New</Button>
        <Button Color="Color.Primary" Disabled="(selectedCountry is null)" Clicked="context.EditCommand.Clicked">Edit</Button>
        <Button Color="Color.Danger" Disabled="(selectedCountry is null)" Clicked="context.DeleteCommand.Clicked">Delete</Button>
        <Button Color="Color.Link" Clicked="context.ClearFilterCommand.Clicked">Clear Filter</Button>
    </ButtonRowTemplate>
    <PopupTitleTemplate>
        Edit
    </PopupTitleTemplate>
</DataGrid>

```
